Added C++11 callbacks and added some cosmetic fixes.

dasmysh opened the pull request:
This should fix #11 .

I also went through the source code and standardized usage of tabs/spaces. Additionally I incorporated some suggestions made by Visual Assist to make use of C++11 features.

alexanderbock commented on include/sgct/SharedData.h:
```diff
@@ -179,10 +183,10 @@ void SharedData::writeVector(SharedVector<T> * vector)
     
     uint32_t element_size = sizeof(T);
-    uint32_t vector_size = static_cast<uint32_t>(tmpVec.size());
+    auto vector_size = static_cast<uint32_t>(tmpVec.size());
```
Same here for `uint32_t`

alexanderbock commented on include/sgct/SharedData.h:
```diff
@@ -200,7 +204,7 @@ void SharedData::readVector(SharedVector<T> * vector)
 
     uint32_t totalSize = size * sizeof(T);
-    unsigned char* data = new unsigned char[ totalSize ];
+    auto* data = new unsigned char[ totalSize ];
```
And here

alexanderbock commented on src/sgct/ClusterManager.cpp:
```diff
@@ -38,15 +38,15 @@ sgct_core::ClusterManager::~ClusterManager()
     nodes.clear();
     
-    for (std::size_t i=0; i < mUsers.size(); i++)
+    for (auto & mUser : mUsers)
```
Cleaner to spell out the type

alexanderbock replied:
better to use `user` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/ClusterManager.cpp:
```diff
@@ -84,22 +84,22 @@ sgct_core::SGCTNode * sgct_core::ClusterManager::getNodePtr(std::size_t index)
 sgct_core::SGCTNode * sgct_core::ClusterManager::getNodePtr(std::string name)
 {
-    for (std::size_t i = 0; i < nodes.size(); i++)
+    for (auto & node : nodes)
```
Cleaner to spell out the type

alexanderbock commented on src/sgct/ClusterManager.cpp:
```diff
@@ -115,29 +115,29 @@ sgct_core::SGCTUser * sgct_core::ClusterManager::getDefaultUserPtr()
 sgct_core::SGCTUser * sgct_core::ClusterManager::getUserPtr(std::string name)
 {
-    for (std::size_t i=0; i < mUsers.size(); i++)
+    for (auto & mUser : mUsers)
```
Cleaner to spell out the type

alexanderbock commented on src/sgct/ClusterManager.cpp:
```diff
@@ -115,29 +115,29 @@ sgct_core::SGCTUser * sgct_core::ClusterManager::getDefaultUserPtr()
 sgct_core::SGCTUser * sgct_core::ClusterManager::getTrackedUserPtr()
 {
-    for (std::size_t i = 0; i < mUsers.size(); i++)
+    for (auto & mUser : mUsers)
```
Cleaner to spell out the type

alexanderbock commented on src/sgct/CorrectionMesh.cpp:
```diff
@@ -91,20 +91,17 @@ sgct_core::CorrectionMeshGeometry::~CorrectionMeshGeometry()
+    mTempIndices = nullptr;
 
-    for (int i = 0; i < LAST_MESH; i++)
+    for (auto & mGeometrie : mGeometries)
```
Cleaner to spell out the type

alexanderbock replied:
Better to call the variable `geometry` ... `m` prefix is designed for member variables

alexanderbock commented on src/sgct/CorrectionMesh.cpp:
```diff
@@ -533,10 +530,10 @@ bool sgct_core::CorrectionMesh::readAndGenerateScalableMesh(const std::string &
-        float xMax = static_cast<float>(orthoCoords[1]);
-        float yMin = static_cast<float>(orthoCoords[2]);
-        float yMax = static_cast<float>(orthoCoords[3]);
+        auto xMin = static_cast<float>(orthoCoords[0]);
```
Better to be explicit and say `float`

alexanderbock commented on src/sgct/CorrectionMesh.cpp:
```diff
@@ -698,7 +695,7 @@ bool sgct_core::CorrectionMesh::readAndGenerateScissMesh(const std::string & mes
     }
     //read vertices
-    SCISSTexturedVertex * texturedVertexList = new SCISSTexturedVertex[numberOfVertices];
+    auto * texturedVertexList = new SCISSTexturedVertex[numberOfVertices];
```
Better to be explicit

alexanderbock commented on src/sgct/CorrectionMesh.cpp:
```diff
@@ -1311,7 +1308,7 @@ bool sgct_core::CorrectionMesh::readAndGeneratePaulBourkeMesh(const std::string
 
     //force regeneration of dome render quad
-    if (FisheyeProjection* fishPrj = dynamic_cast<FisheyeProjection*>(parent->getNonLinearProjectionPtr()))
+    if (auto* fishPrj = dynamic_cast<FisheyeProjection*>(parent->getNonLinearProjectionPtr()))
```
Sticking with the explicit type might be better

alexanderbock commented on src/sgct/SGCTNetwork.cpp:
```diff
@@ -480,7 +480,7 @@ void sgct_core::SGCTNetwork::pushClientMessage()
 
         uint32_t dataSize = currentMessageSize - mHeaderSize;
-        unsigned char *currentMessageSizePtr = reinterpret_cast<unsigned char *>(&dataSize);
+        auto *currentMessageSizePtr = reinterpret_cast<unsigned char *>(&dataSize);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTNetwork.cpp:
```diff
@@ -505,7 +505,7 @@ void sgct_core::SGCTNetwork::pushClientMessage()
 
         uint32_t localSyncHeaderSize = 0;
-        unsigned char *currentMessageSizePtr = (unsigned char *)&localSyncHeaderSize;
+        auto *currentMessageSizePtr = (unsigned char *)&localSyncHeaderSize;
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTNetwork.cpp:
```diff
@@ -929,7 +929,7 @@ int sgct_core::SGCTNetwork::readExternalMessage()
 void sgct_core::SGCTNetwork::communicationHandlerStarter(void *arg)
 {
-    sgct_core::SGCTNetwork * nPtr = (sgct_core::SGCTNetwork *)arg;
+    auto * nPtr = (sgct_core::SGCTNetwork *)arg;
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTNetwork.cpp:
```diff
@@ -1088,7 +1088,7 @@ void sgct_core::SGCTNetwork::communicationHandler()
                         {
                             //parse the package id
-                            uLongf uncompressedSize = static_cast<uLongf>(uncompressedDataSize);
+                            auto uncompressedSize = static_cast<uLongf>(uncompressedDataSize);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTNetwork.cpp:
```diff
@@ -1262,7 +1262,7 @@ void sgct_core::SGCTNetwork::communicationHandler()
                         {
                             //parse the package id
-                            uLongf uncompressedSize = static_cast<uLongf>(uncompressedDataSize);
+                            auto uncompressedSize = static_cast<uLongf>(uncompressedDataSize);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTNetwork.cpp:
```diff
@@ -1287,8 +1287,8 @@ void sgct_core::SGCTNetwork::communicationHandler()
                             uint32_t pLenght = 0;
-                            char *packageIdPtr = reinterpret_cast<char *>(&packageId);
-                            char *sizeDataPtr = reinterpret_cast<char *>(&pLenght);
+                            auto *packageIdPtr = reinterpret_cast<char *>(&packageId);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTNode.cpp:
```diff
@@ -47,8 +47,8 @@ bool sgct_core::SGCTNode::getKeyPressed( int key )
 
-    for(std::size_t i=0; i<mWindows.size(); i++)
-        if( glfwGetKey( mWindows[i].getWindowHandle(), key) )
+    for(auto & mWindow : mWindows)
```
Better to be explicit in the type
Better to use `window` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTNode.cpp:
```diff
@@ -59,15 +59,15 @@ bool sgct_core::SGCTNode::getKeyPressed( int key )
     std::size_t counter = 0;
-    for(std::size_t i=0; i<mWindows.size(); i++)
-        if( glfwWindowShouldClose( mWindows[i].getWindowHandle() ) )
+    for(auto & mWindow : mWindows)
```
Better to be explicit in the type
Better to use `window` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTNode.cpp:
```diff
@@ -59,15 +59,15 @@ bool sgct_core::SGCTNode::getKeyPressed( int key )
 
-    for(std::size_t i=0; i<mWindows.size(); i++)
-    if (!(mWindows[i].isVisible() || mWindows[i].isRenderingWhileHidden()))
+    for(auto & mWindow : mWindows)
```
Better to be explicit in the type
Better to use `window` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTNode.cpp:
```diff
@@ -81,8 +81,8 @@ bool sgct_core::SGCTNode::shouldAllWindowsClose()
 {
-    for(std::size_t i=0; i<mWindows.size(); i++)
-        mWindows[i].setVisibility( true );
+    for(auto & mWindow : mWindows)
```
Better to be explicit in the type
Better to use `window` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTNode.cpp:
```diff
@@ -98,8 +98,8 @@ bool sgct_core::SGCTNode::isUsingSwapGroups()
 {
-    for(std::size_t i=0; i<mWindows.size(); i++)
-        mWindows[i].setVisibility( false );
+    for(auto & mWindow : mWindows)
```
Better to be explicit in the type
Better to use `window` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTSettings.cpp:
```diff
@@ -45,8 +45,8 @@ sgct::SGCTSettings::SGCTSettings()
 
-    for(size_t i=0; i<3; i++)
-        mCapturePath[i].assign("SGCT");
+    for(auto & path : mCapturePath)
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTTracker.cpp:
```diff
@@ -23,12 +23,12 @@ sgct::SGCTTracker::SGCTTracker(std::string name)
 sgct::SGCTTracker::~SGCTTracker()
 {
-    for(size_t i=0; i<mTrackingDevices.size(); i++)
+    for(auto & mTrackingDevice : mTrackingDevices)
```
Better to be explicit in the type
Better to use `trackingDevice` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTTracker.cpp:
```diff
@@ -37,9 +37,9 @@ sgct::SGCTTracker::~SGCTTracker()
 void sgct::SGCTTracker::setEnabled(bool state)
 {
-    for(size_t i=0; i<mTrackingDevices.size(); i++)
+    for(auto & mTrackingDevice : mTrackingDevices)
```
Better to be explicit in the type
Better to use `trackingDevice` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTTracker.cpp:
```diff
@@ -54,35 +54,35 @@ void sgct::SGCTTracker::addDevice(std::string name, size_t index)
 sgct::SGCTTrackingDevice * sgct::SGCTTracker::getDevicePtrBySensorId(int id)
 {
-    for(size_t i=0; i<mTrackingDevices.size(); i++)
+    for(auto & mTrackingDevice : mTrackingDevices)
```
Better to be explicit in the type
Better to use `trackingDevice` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -148,11 +148,11 @@ void sgct::SGCTTrackingManager::startSampling()
 {
-    for(size_t i=0; i<mTrackers.size(); i++)
-        for(size_t j=0; j<mTrackers[i]->getNumberOfDevices(); j++)
+    for(auto & mTracker : mTrackers)
```
Better to be explicit in the type
Better to use `tracker` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -284,18 +284,18 @@ void sgct::SGCTTrackingManager::setHeadTracker(const char * trackerName, const c
 void samplingLoop(void *arg)
 {
-    sgct::SGCTTrackingManager * tmPtr =
+    auto * tmPtr =
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -338,33 +338,33 @@ void samplingLoop(void *arg)
 sgct::SGCTTracker * sgct::SGCTTrackingManager::getTrackerPtr(const char * name)
 {
-    for(size_t i=0; i<mTrackers.size(); i++)
+    for(auto & mTracker : mTrackers)
```
Better to be explicit in the type
Better to use `tracker` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -393,14 +393,14 @@ double sgct::SGCTTrackingManager::getSamplingTime()
 void VRPN_CALLBACK update_tracker_cb(void *userdata, const vrpn_TRACKERCB info)
 {
-    sgct::SGCTTracker * trackerPtr =
+    auto * trackerPtr =
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -338,33 +338,33 @@ void samplingLoop(void *arg)
 void sgct::SGCTTrackingManager::setEnabled(bool state)
 {
-    for(size_t i=0; i<mTrackers.size(); i++)
+    for(auto & mTracker : mTrackers)
```
Better to be explicit in the type
Better to use `tracker` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -413,7 +413,7 @@ void VRPN_CALLBACK update_tracker_cb(void *userdata, const vrpn_TRACKERCB info)
 void VRPN_CALLBACK update_button_cb(void *userdata, const vrpn_BUTTONCB b )
 {
-    sgct::SGCTTrackingDevice * devicePtr =
+    auto * devicePtr =
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTTrackingManager.cpp:
```diff
@@ -425,7 +425,7 @@ void VRPN_CALLBACK update_button_cb(void *userdata, const vrpn_BUTTONCB b )
 void VRPN_CALLBACK update_analog_cb(void* userdata, const vrpn_ANALOGCB a )
 {
-    sgct::SGCTTrackingDevice * tdPtr =
+    auto * tdPtr =
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTUser.cpp:
```diff
@@ -16,8 +16,8 @@ sgct_core::SGCTUser::SGCTUser(std::string name)
     
-    for(unsigned int i=0; i<3; i++)
-        mPos[i] = glm::vec3(0.0f);
+    for(auto & mPo : mPos)
```
Better to be explicit in the type
Better to use `pos` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTWindow.cpp:
```diff
@@ -223,29 +223,29 @@ void sgct::SGCTWindow::close()
     MessageHandler::instance()->print(MessageHandler::NOTIFY_INFO, "Deleting screen capture data for window %d...\n", mId);
-    for (int i = 0; i < 2; i++)
-        if( mScreenCapture[i] )
+    for (auto & i : mScreenCapture)
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTWindow.cpp:
```diff
@@ -339,17 +339,17 @@ void sgct::SGCTWindow::initOGL()
 
-    for (std::size_t i = 0; i < mViewports.size(); i++)
-        if (mViewports[i]->hasSubViewports())
+    for (auto & mViewport : mViewports)
```
Better to be explicit in the type
Better to use `viewport` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTWindow.cpp:
```diff
@@ -594,29 +594,29 @@ bool sgct::SGCTWindow::update()
         //resize PBOs
-        for (int i = 0; i < 2; i++)
-            if (mScreenCapture[i] != NULL)
+        for (auto & i : mScreenCapture)
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTWindow.cpp:
```diff
@@ -594,29 +594,29 @@ bool sgct::SGCTWindow::update()
         //resize non linear projection buffers
-        for (std::size_t i = 0; i < mViewports.size(); i++)
-            if (mViewports[i]->hasSubViewports())
+        for (auto & mViewport : mViewports)
```
Better to be explicit in the type
Better to use `viewport` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/SGCTWindow.cpp:
```diff
@@ -1642,7 +1642,7 @@ const sgct::SGCTWindow::StereoMode & sgct::SGCTWindow::getStereoMode() const
 void sgct::SGCTWindow::addViewport(float left, float right, float bottom, float top)
 {
-    sgct_core::Viewport * vpPtr = new sgct_core::Viewport(left, right, bottom, top);
+    auto * vpPtr = new sgct_core::Viewport(left, right, bottom, top);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SGCTWindow.cpp:
```diff
@@ -1658,10 +1658,10 @@ void sgct::SGCTWindow::addViewport(sgct_core::Viewport * vpPtr)
 
-    for (unsigned int i = 0; i < mViewports.size(); i++)
-        delete mViewports[i];
+    for (auto & mViewport : mViewports)
```
Better to be explicit in the type
Better to use `viewport` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/ScreenCapture.cpp:
```diff
@@ -222,36 +222,36 @@ void sgct_core::ScreenCapture::saveScreenCapture(unsigned int textureId, CaputeS
             glPopAttrib();
         
-        GLubyte * ptr = reinterpret_cast<GLubyte*>(glMapBuffer(GL_PIXEL_PACK_BUFFER, GL_READ_ONLY));
+        auto * ptr = reinterpret_cast<GLubyte*>(glMapBuffer(GL_PIXEL_PACK_BUFFER, GL_READ_ONLY));
```
Better to be explicit in the type

alexanderbock commented on src/sgct/ScreenCapture.cpp:
```diff
@@ -519,7 +519,7 @@ sgct_core::Image * sgct_core::ScreenCapture::prepareImage(int index)
 void screenCaptureHandler(void *arg)
 {
-    sgct_core::ScreenCaptureThreadInfo * ptr = reinterpret_cast<sgct_core::ScreenCaptureThreadInfo *>(arg);
+    auto * ptr = reinterpret_cast<sgct_core::ScreenCaptureThreadInfo *>(arg);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/Shader.cpp:
```diff
@@ -189,9 +187,9 @@ bool sgct_core::Shader::checkCompilationStatus() const
         }
 
-        GLchar * log = new GLchar[logLength];
+        auto * log = new GLchar[logLength];
```
Better to be explicit in the type

alexanderbock commented on src/sgct/ShaderManager.cpp:
```diff
@@ -28,19 +28,19 @@ sgct::ShaderManager::ShaderManager(void) :
+sgct::ShaderManager::~ShaderManager()
 {
-    for(unsigned int i=0; i < NUMBER_OF_SHADER_BINS; i++)
+    for(auto & mShaderProgram : mShaderPrograms)
```
Better to be explicit in the type

alexanderbock commented on src/sgct/ShaderProgram.cpp:
```diff
@@ -54,12 +54,12 @@ Will deattach all attached shaders, delete them and then delete the program
 void sgct::ShaderProgram::deleteProgram()
 {        
-    for(std::size_t i=0; i<mShaders.size(); i++)
+    for(auto & mShader : mShaders)
```
Better to be explicit in the type
Better to use `shader` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/ShaderProgram.cpp:
```diff
@@ -163,10 +163,10 @@ bool sgct::ShaderProgram::createAndLinkProgram()
     //
-    for(std::size_t i=0; i<mShaders.size(); i++)
-        if( mShaders[i].mShader.getId() > GL_FALSE )
+    for(auto & mShader : mShaders)
```
Better to be explicit in the type
Better to use `shader` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/ShaderProgram.cpp:
```diff
@@ -185,17 +185,17 @@ bool sgct::ShaderProgram::reload()
     deleteProgram();
 
-    for(std::size_t i=0; i<mShaders.size(); i++)
+    for(auto & mShader : mShaders)
```
Better to be explicit in the type
Better to use `shader` , `m` prefix is used for member variables

alexanderbock commented on src/sgct/ShaderProgram.cpp:
```diff
@@ -250,8 +250,8 @@ bool sgct::ShaderProgram::checkLinkStatus() const
 
-        GLchar * log = new GLchar[logLength];
-        glGetProgramInfoLog( mProgramId, logLength, NULL, log );
+        auto * log = new GLchar[logLength];
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -198,8 +234,8 @@ void SharedData::encode()
 
-        uLongf compressed_size = static_cast<uLongf>(mCompressedBufferSize);
-        uLongf data_size = static_cast<uLongf>(dataBlockToCompress.size());
+        auto compressed_size = static_cast<uLongf>(mCompressedBufferSize);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -210,8 +246,8 @@ void SharedData::encode()
             //add original size
-            uint32_t uncompressedSize = static_cast<uint32_t>(dataBlockToCompress.size());
-            unsigned char *p = reinterpret_cast<unsigned char *>(&uncompressedSize);
+            auto uncompressedSize = static_cast<uint32_t>(dataBlockToCompress.size());
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -247,7 +283,7 @@ void SharedData::writeFloat(SharedFloat * sf)
     float val = sf->getVal();
     SGCTMutexManager::instance()->lockMutex( sgct::SGCTMutexManager::DataSyncMutex );
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -260,7 +296,7 @@ void SharedData::writeDouble(SharedDouble * sd)
     double val = sd->getVal();
     SGCTMutexManager::instance()->lockMutex( sgct::SGCTMutexManager::DataSyncMutex );
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -273,7 +309,7 @@ void SharedData::writeInt64(SharedInt64 * si)
     int64_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex( sgct::SGCTMutexManager::DataSyncMutex );
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -286,7 +322,7 @@ void SharedData::writeInt32(SharedInt32 * si)
     int32_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -299,7 +335,7 @@ void SharedData::writeInt16(SharedInt16 * si)
     int16_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -312,7 +348,7 @@ void SharedData::writeInt8(SharedInt8 * si)
     int8_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -325,7 +361,7 @@ void SharedData::writeUInt64(SharedUInt64 * si)
     uint64_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -338,7 +374,7 @@ void SharedData::writeUInt32(SharedUInt32 * si)
     uint32_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -351,7 +387,7 @@ void SharedData::writeUInt16(SharedUInt16 * si)
     uint16_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -364,7 +400,7 @@ void SharedData::writeUInt8(SharedUInt8 * si)
     uint8_t val = si->getVal();
     SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -404,8 +440,8 @@ void SharedData::writeString(SharedString * ss)
     SGCTMutexManager::instance()->lockMutex( sgct::SGCTMutexManager::DataSyncMutex );
-    uint32_t length = static_cast<uint32_t>(tmpStr.size());
-    unsigned char *p = reinterpret_cast<unsigned char *>(&length);
+    auto length = static_cast<uint32_t>(tmpStr.size());
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -416,19 +452,19 @@ void SharedData::writeString(SharedString * ss)
-	unsigned char *ws = reinterpret_cast<unsigned char *>(&tmpStr[0]);
+    std::wstring tmpStr(ss->getVal());
+    SGCTMutexManager::instance()->lockMutex(sgct::SGCTMutexManager::DataSyncMutex);
+    auto length = static_cast<uint32_t>(tmpStr.size());
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -448,7 +484,7 @@ void SharedData::writeSize(uint32_t size)
     
     SGCTMutexManager::instance()->lockMutex( sgct::SGCTMutexManager::DataSyncMutex );
-    unsigned char *p = reinterpret_cast<unsigned char *>(&size);
+    auto *p = reinterpret_cast<unsigned char *>(&size);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SharedData.cpp:
```diff
@@ -674,7 +710,7 @@ void SharedData::readString(SharedString * ss)
     }
 
-    char * stringData = new (std::nothrow) char[length+1];
+    auto * stringData = new (std::nothrow) char[length+1];
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SphericalMirrorProjection.cpp:
```diff
@@ -108,7 +105,7 @@ void sgct_core::SphericalMirrorProjection::initTextures()
 void sgct_core::SphericalMirrorProjection::initVBO()
 {
-    if (Viewport * vp = dynamic_cast<Viewport*>(sgct::Engine::instance()->getCurrentWindowPtr()->getCurrentViewport()))
+    if (auto * vp = dynamic_cast<Viewport*>(sgct::Engine::instance()->getCurrentWindowPtr()->getCurrentViewport()))
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SphericalMirrorProjection.cpp:
```diff
@@ -468,7 +465,7 @@ void sgct_core::SphericalMirrorProjection::renderCubemapInternal(std::size_t * s
         *subViewPortIndex = i;
         faceIndex = static_cast<unsigned int>(i);
-        TextureIndex ti = static_cast<TextureIndex>(CubeFaceRight + i);
+        auto ti = static_cast<TextureIndex>(CubeFaceRight + i);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/SphericalMirrorProjection.cpp:
```diff
@@ -496,7 +493,7 @@ void sgct_core::SphericalMirrorProjection::renderCubemapInternalFixedPipeline(st
         *subViewPortIndex = i;
         faceIndex = static_cast<unsigned int>(i);
-        TextureIndex ti = static_cast<TextureIndex>(CubeFaceRight + i);
+        auto ti = static_cast<TextureIndex>(CubeFaceRight + i);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/Statistics.cpp:
```diff
@@ -406,7 +406,7 @@ void sgct_core::Statistics::draw(float lineWidth)
         glPushMatrix();
         
-        double size = static_cast<double>(STATS_HISTORY_LENGTH);
+        auto size = static_cast<double>(STATS_HISTORY_LENGTH);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/Statistics.cpp:
```diff
@@ -510,7 +510,7 @@ void sgct_core::Statistics::draw(float lineWidth)
         glLineWidth( lineWidth );
         
-        float size = static_cast<float>(STATS_HISTORY_LENGTH);
+        auto size = static_cast<float>(STATS_HISTORY_LENGTH);
```
Better to be explicit in the type

alexanderbock commented on src/sgct/TextureManager.cpp:
```diff
@@ -367,7 +367,7 @@ bool sgct::TextureManager::uploadImage(sgct_core::Image * imgPtr, unsigned int *
 
     GLint internalFormat;
-    unsigned int bpc = static_cast<unsigned int>(imgPtr->getBytesPerChannel());
+    auto bpc = static_cast<unsigned int>(imgPtr->getBytesPerChannel());
```
Better to be explicit in the type

alexanderbock commented on src/sgct/TextureManager.cpp:
```diff
@@ -487,8 +487,8 @@ void sgct::TextureManager::freeTextureData()
-    for (auto it = mTextures.begin(); it != mTextures.end(); ++it)
-        if (it->second.mId)
-            glDeleteTextures(1, &(it->second.mId));
+    for (auto & mTexture : mTextures)
```
Better to be explicit in the type

alexanderbock commented on src/sgct/Viewport.cpp:
```diff
@@ -344,58 +344,58 @@ void sgct_core::Viewport::parseFisheyeProjection(tinyxml2::XMLElement * element)
-
-	spoutProj->setUseDepthTransformation(true);
-	mNonLinearProjection = spoutProj;
+    auto * spoutProj = new SpoutOutputProjection();
```
Better to be explicit in the type

alexanderbock commented on src/sgct/Viewport.cpp:
```diff
@@ -344,58 +344,58 @@ void sgct_core::Viewport::parseFisheyeProjection(tinyxml2::XMLElement * element)
 void sgct_core::Viewport::parseSphericalMirrorProjection(tinyxml2::XMLElement * element)
 {
-    SphericalMirrorProjection * sphericalMirrorProj = new SphericalMirrorProjection();
+    auto * sphericalMirrorProj = new SphericalMirrorProjection();
```
Better to be explicit in the type

alexanderbock commented on src/sgct/freetype.cpp:
```diff
@@ -129,467 +129,467 @@ inline void pop_projection_matrix()
-	if (buffer == NULL)
-		return NULL;
+    int size = 1 + vscprintf(format, args);
+    auto * buffer = new (std::nothrow) char[size];
```
Better to be explicit in the type

alexanderbock commented on src/sgct/freetype.cpp:
```diff
@@ -129,467 +129,467 @@ inline void pop_projection_matrix()
-	if (buffer == NULL)
-		return NULL;
+    int size = 1 + vscwprintf(format, args);
+    auto * buffer = new (std::nothrow) wchar_t[size];
```
Better to be explicit in the type

dasmysh commented:
I see the problem with the prefix of names in the for-loops and you can also argue about the use of `auto` in these cases (I do not agree though). I will fix these issues.

All the other cases for `auto` usage (and I've really reviewed all of them) are explicit in type either due to cases where the type is explicitly states or due to `new` operators where the type is also explicitly stated. Using auto in these cases does not remove the explicit type statement but just adds redundancy.

patlj09 commented:
Hi Alex,

Just a question, I believe I read something about this, but forgot.

What’s the difference between:

    auto *p = reinterpret_cast<unsigned char *>(&val);


versus no explicit pointer with auto

    auto p = reinterpret_cast<unsigned char *>(&val);


?

/ Patric

On 29 Sep 2017, at 12:39, Alexander Bock <notifications@github.com<mailto:notifications@github.com>> wrote:


@alexanderbock commented on this pull request.

________________________________

In include/sgct/SharedData.h<https://github.com/opensgct/sgct/pull/18#discussion_r141950896>:

> @@ -157,7 +161,7 @@ void SharedData::writeObj( SharedObject<T> * sobj )
     T val = sobj->getVal();

     SGCTMutexManager::instance()->lockMutex(SGCTMutexManager::DataSyncMutex);
-    unsigned char *p = reinterpret_cast<unsigned char *>(&val);
+    auto *p = reinterpret_cast<unsigned char *>(&val);


Keeping with the unsigned char * might be cleaner here

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/opensgct/sgct/pull/18#pullrequestreview-66251259>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AOL3QPki1DsiygG_tyK8EJVKsmhTR_bzks5snUd3gaJpZM4PoWVl>.

alexanderbock commented:
@dasmysh 

I really believe using the `auto` keyword just to reduce redundancy is a bit misleading... A code will be read many many more times than it will be written.  And everytime when someone reads the code, they will parse two expressions to understand the type of an object rather than just one.  But I'm fine with having it in those cases where the type is named explicitly on the rhs and I'll change it later.  With the for loops, it only hinders the understanding of the code and it is important to spell out the type for cases where a non std-container or function is used (as an example,  that it's fine to have `auto` as a replacement for `std::map<int, std::string, std::more<int>, std::allocator<std::pair<const int, std::string>>>::const_iterator` as people can be expected to be familiar with the STL)


@patlj09  There is no difference.  In the top case `auto` deduces to `unsigned char`, where in the bottom part, `auto` deduces to `unsigned char*`.  The difference comes into play with references and  cv qualifiers.  
 § 7.1.6.4    http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf

dasmysh commented:
Sorry this took me so long.

I updated to the current master branch and removed the conflicts that were there.
After that I tried to change all the parts of the code mentioned:
- I added types to the range based for loops that had auto before.
- I fixed the auto generated variable names to have no member prefix and (in some cases) meaningful names.

I hope I got all cases.

There are some warnings in CorrectionMesh.cpp that I did not fix to not make this merge even more cluttered.
